app/src/main/java/com/jerboa/db/AppDB.kt

```diff
@@ -142,8 +149,30 @@ val APP_SETTINGS_DEFAULT = AppSettings(
     usePrivateTabs = false,
     secureWindow = false,
     blurNSFW = true,
+    saveSearchHistory = true,
+)
+
+@Entity
+data class SearchHistory(
+    @PrimaryKey @ColumnInfo(name = "text") val text: String,
```

Comment:
No, use auto-gen'd primary keys like the other example.

2ndly, don't name these columns names which might be protected keywords in sql or kotlin. Name it `search_term` or something.

3rd: These search terms should almost certainly be attached to an account, so have a foreign key link to account.id